docs/proposals/parallel-compaction.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Parallel Compaction by Time Interval"
+linkTitle: "Parallel Compaction by Time Interval"
+weight: 1
+slug: parallel-compaction
+---
+
+- Author: [Roy Chiang](https://github.com/roystchiang)
+- Date: May 2021
+- Status: Proposed
+---
+
+## Introduction
+As a part of pushing Cortex’s scaling capability at AWS, we have done performance testing with Cortex and found the compactor to be one of the main limiting factors for higher active timeseries limit per tenant. The documentation [Compactor](https://cortexmetrics.io/docs/blocks-storage/compactor/#how-compaction-works) describes the responsibilities of a compactor, and this proposal focuses on the limitations of the current compactor architecture. In the current architecture, compactor has simple sharding, meaning that a single tenant is sharded to a single compactor. In addition, a compactor handles compaction groups of a single tenant iteratively, meaning that blocks belonging non-overlapping times are not compacted in parallel.
```

Comment:
"compaction groups" is not defined before use, either in this document or at the referenced link.
Possibly it could go in https://github.com/cortexproject/cortex/blob/master/docs/guides/glossary.md